tests/test_jwt_object.cc

```diff
@@ -0,0 +1,35 @@
+#include "gtest/gtest.h"
+#include "jwt/jwt.hpp"
+
+namespace {
+
+struct Wrapper
+{
+    // The std::move here is required to resolve to the move ctor
+    // rather than to the universal reference ctor.
+    Wrapper(jwt::jwt_object&& obj) : object{std::move(obj)} {}
+    jwt::jwt_object object;
+};
+
+} // END namespace
+
+TEST (ObjectTest, MoveConstructor)
+{
+  using namespace jwt::params;
+
+  jwt::jwt_object obj{algorithm("hs256"), secret("secret")};
+
+  obj.add_claim("iss", "arun.muralidharan");
+
+  auto wrapper = Wrapper{std::move(obj)};
```

Comment:
Hmm...so a wrapper kind of thing is required to make the move work ? Would it be better if we add a SFINAE check to make the constructor less greedy ?

Reply:
The wrapper is not necessary to use the move constructor. However, it *was* necessary to reproduce the compiler error I was seeing (GCC 7.3). If you remove the `std::move` on line 10, you can see it. I wasn't able to reproduce the error with a simpler example. I recognize the limited value of this test, so I don't mind if you reject the PR. (Tests of things that are not supposed to compile, such as passing an object that does not model `ParameterConcept`, are tricky.)

Reply:
As for SFINAE on the universal reference ctor, maybe this?
```
  template <typename First, typename... Rest>
  jwt_object(std::enable_if_t<detail::meta::is_parameter_concept<First>::value, First>&& first, Rest&&... rest);
```

Reply:
Hmm...yup that looks about right.

Reply:
Would you like to provide a PR for this change as well ?

Reply:
Pushed a commit to this PR with the new SFINAE guard.